<p align="center">
  <img height="100" src="AWmeta_logo.jpg" />
</p>


[![stars-badge](https://img.shields.io/github/stars/YanshiHu/AWmeta)](https://github.com/YanshiHu/AWmeta/stargazers)
![](https://img.shields.io/badge/language-R4.1.0-orange.svg)
![](https://img.shields.io/badge/environment-conda-blue.svg)
![](https://img.shields.io/badge/verison-1.0.0-ff69b4.svg)
[![Preprint](https://img.shields.io/badge/preprint-bioRœáiv-C93A41)](https://www.biorxiv.org/content/10.1101/2025.05.06.650408)
[![](https://img.shields.io/badge/DOI-10.1101/2025.05.06.650408-4270C6)](https://doi.org/10.1101/2025.05.06.650408)
[![license-badge](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)


## **[<b>AWmeta</b>](https://github.com/YanshiHu/AWmeta) ([<ins><b>A</b></ins>](https://github.com/YanshiHu/AWmeta)daptively-[<ins><b>W</b></ins>](https://github.com/YanshiHu/AWmeta)eighted transcriptomic [<ins><b>meta</b></ins>](https://github.com/YanshiHu/AWmeta)-analysis, […îÀêÀàm…õt…ô])**

> [!IMPORTANT]
> We will release scripts soon, please stay tuned. :)

<p align="justify">
    Transcriptomic meta-analysis, a statistical method to synthesize multiple independent studies on a common topic, significantly enhances statistical power and facilitates the identification of consistent molecular characteristics, thereby improving reliability and reproducibility of biological discoveries. While many meta-analysis methods and tools have been developed for transcriptomic data, each manifests certain limitations, typically relying on either <i>P</i>-value combination or effect size integration alone. To address these limitations, we introduce AWmeta, a novel method for transcriptomic meta-analysis, which, for the first time, integrates the advantages of transcriptomic meta-analysis <i>P</i>-value and effect size methods, resulting in performance enhancement. By leveraging 35 transcriptomic datasets from Parkinson's and Crohn's disease, encompassing diverse tissue types, AWmeta exerts superior performance over existing state-of-the-art methods, including REM (Random Effect Model), across a range of key metrics. Specifically, AWmeta provides highly reliable differentially expressed genes (DEGs) via accurate DEG discrimination in real and simulated data. Moreover, AWmeta results in gene difference quantification converging closer to theoretical baselines at both gene- and study-level compared to independent studies and REM. Importantly, AWmeta requires only a small sample size of independent studies to achieve substantial gene difference quantification effects, aiding in reducing experimental costs. We also observed that AWmeta requires fewer DEGs from integrated independent studies to achieve significant gene difference quantification effects, easing experimental operation and reducing researcher workload. While maintaining reliable differential gene quantification, AWmeta demonstrates robust performance against external interference and internal defects. In addition, AWmeta generates gene difference quantification results that offer more biologically meaningful insights than those from original independent studies and REM. Systematic evaluation demonstrates that AWmeta outperforms existing methods by achieving superior convergence, stability, and robustness with smaller sample size and fewer DEGs. Overall, AWmeta serves as a powerful and effective tool for transcriptomic meta-analysis.
</p>


## Overview

<img src="AWmeta_methodology.jpg" width="850">


## ‚úçÔ∏è Citation<a name="citation"></a>

**To cite AWmeta:**

<ins>Yanshi Hu</ins><sup>$</sup>, Zixuan Wang, Yueming Hu, Cong Feng, Qiuyu Fang, Ming Chen<sup>$</sup> (2025). [AWmeta empowers adaptively-weighted transcriptomic meta-analysis](https://doi.org/10.1101/2025.05.06.650408). <i><b>bio<span style="color:#ff0000;">R</span>&chi;iv</b></i>, 2025.05.06.650408.


## üòµ Issues<a name="issues"></a>

If you have any questions or thoughts, please feel free to  [file an issue](https://github.com/YanshiHu/AWmeta/issues) or direct to yanshihu [at] zju [dot] edu [dot] cn.

