<p align="center">
  <img height="100" src="AWmeta_logo.jpg" />
</p>


[![stars-badge](https://img.shields.io/github/stars/YanshiHu/AWmeta)](https://github.com/YanshiHu/AWmeta/stargazers)
![](https://img.shields.io/badge/language-R4.1.0-orange.svg)
![](https://img.shields.io/badge/environment-conda-blue.svg)
![](https://img.shields.io/badge/verison-1.0.0-ff69b4.svg)
[![Preprint](https://img.shields.io/badge/preprint-bioRœáiv-C93A41)](https://www.biorxiv.org/content/10.1101/2025.05.06.650408)
[![](https://img.shields.io/badge/DOI-10.1101/2025.05.06.650408-4270C6)](https://doi.org/10.1101/2025.05.06.650408)
[![license-badge](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)


## **[<b>AWmeta</b>](https://github.com/YanshiHu/AWmeta) ([<ins><b>A</b></ins>](https://github.com/YanshiHu/AWmeta)daptively-[<ins><b>W</b></ins>](https://github.com/YanshiHu/AWmeta)eighted transcriptomic [<ins><b>meta</b></ins>](https://github.com/YanshiHu/AWmeta)-analysis, […îÀêÀàm…õt…ô])**

> [!IMPORTANT]
> We will release scripts soon, please stay tuned. :)

<p align="justify">
Transcriptomic meta-analysis enhances biological discovery reproducibility by integrating cross-study data, yet existing methods rely exclusively on <i>P</i>-value or effect size integration. We present AWmeta, an adaptively-weighted framework unifying both paradigms. Benchmarking across 35 Parkinson‚Äôs and Crohn‚Äôs disease datasets spanning diverse tissues, AWmeta exerts superior performance over state-of-the-art random effects model (REM), across a range of key metrics. Specifically, AWmeta identifies highly reliable differentially expressed genes (DEGs) in real and simulated settings. Moreover, AWmeta results in gene- and study-wise gene differential quantification closer to theoretical baselines compared to REM and original studies. Importantly, AWmeta requires smaller sample size and fewer DEGs of original studies to achieve substantial gene differential quantification, aiding in reducing experimental costs. While maintaining reliable gene differential quantification, AWmeta demonstrates robustness against external interference and internal defects. Crucially, AWmeta prioritizes disease tissue-specific mechanisms with higher functional coherence than those from REM and original studies. By bridging statistical rigor with mechanistic interpretability, AWmeta harmonizes heterogeneous transcriptomic data into actionable insights, serving as a transformative tool for precision transcriptomic integration.
</p>


## Overview

<img src="AWmeta_methodology.jpg" width="850">


## ‚úçÔ∏è Citation<a name="citation"></a>

**To cite AWmeta:**

<ins>Yanshi Hu</ins><sup>$</sup>, Zixuan Wang, Yueming Hu, Cong Feng, Qiuyu Fang, Ming Chen<sup>$</sup> (2025). [AWmeta empowers adaptively-weighted transcriptomic meta-analysis](https://doi.org/10.1101/2025.05.06.650408). <i><b>bio<span style="color:#ff0000;">R</span>&chi;iv</b></i>, 2025.05.06.650408.


## üòµ Issues<a name="issues"></a>

If you have any questions or thoughts, please feel free to  [file an issue](https://github.com/YanshiHu/AWmeta/issues) or direct to yanshihu [at] zju [dot] edu [dot] cn.

